packages/admin-panel/src/VizBuilderApp/components/DataLibrary/AggregationDataLibrary.js

```diff
@@ -17,24 +17,43 @@ const aggregateToValue = aggregate =>
 
 const valueToAggregate = value => value.map(({ code, config }) => ({ type: code, config }));
 
-export const AggregationDataLibrary = ({ aggregate, onAggregateChange }) => {
+export const AggregationDataLibrary = ({ aggregate, onAggregateChange, onInvalidChange }) => {
   const [inputValue, setInputValue] = useState('');
-  const debouncedInputValue = useDebounce(inputValue, 200);
   const value = aggregateToValue(aggregate);
-  const { data: aggregationOptionSearchResults, isFetching } = useSearchAggregationOptions({
-    search: debouncedInputValue,
-  });
-  const options = inputValue ? aggregationOptionSearchResults : [];
+  const { data: options, isFetching } = useSearchAggregationOptions();
+
+  const onChange = (event, newValue) => onAggregateChange(valueToAggregate(newValue));
+  const onRemove = (event, option) => {
+    onChange(
+      event,
+      value.filter(item => option.code !== item.code),
+    );
+  };
 
   return (
     <DataLibrary
-      options={options}
+      options={options || []}
       value={value}
-      onChange={(event, newValue) => onAggregateChange(valueToAggregate(newValue))}
+      onChange={onChange}
       inputValue={inputValue}
       onInputChange={(event, newInputValue) => (event ? setInputValue(newInputValue) : false)}
       isLoading={isFetching}
       searchPageSize={MAX_RESULTS}
+      onMouseEnter={prefetchAggregationOptions}
+      optionComponent={(option, index, setEdittingOption) => (
+        <SelectedOptionWithJsonEditor
+          option={option}
+          optionMetaData={options && options.find(({ code }) => code === option.code)}
+          onChange={newValue => {
+            const newSelectedAggregations = Array.from(value);
+            newSelectedAggregations[index].config = newValue.config;
+            onAggregateChange(valueToAggregate(newSelectedAggregations));
+          }}
+          onRemove={onRemove}
```

Comment:
Extract `onRemove` from DataLibrary to AggregationDataLibrary because `filter` function is different.